Classes/Extensions/Amazing+Tools.swift

```diff
@@ -26,181 +25,317 @@ public extension Amazing {
     
 }
 
-
-extension Awesome.brand {
-
-    public typealias AwesomeType = Awesome.brand
-
-    public static func fromType(_ code: String) -> AwesomeType? {
-        guard let icon = AwesomeType(rawValue: code) else {
-            return nil
-        }
-        return icon
+extension Awesome.Brand {
+    
+    public var code: String {
+        return self.rawValue
     }
-
+    
+    public var key: String {
+        return String(describing: self)
+    }
+    
+    public var description: String {
+        return String(describing: self).split(separator: "-").map { $0.firstUppercased }.joined(separator: " ")
```

Comment:
What are you trying to achieve with this actually? There are no hyphens in the keys therefore this does essentially nothing.

`Awesome.Solid.americanSignLanguageInterpreting.label` printed "American Sign Language Interpreting" before and now it prints "AmericanSignLanguageInterpreting" therefore `description` lost value compared to before.

Same thing for the key. Before it contained hyphens now it does not.

Not sure how to store the descriptions and keys.

But here are several possibilities

* Store them in an array as before
* Store them in a dictionary
* Generate a `switch` which returns the specific label / name (description / key)
    - `-` Makes the files very long
* Make the RawValue a custom type which stores all needed values (unicode, name and label)
    - `+` Shorter enum files
    - `-` Makes the RawValue inits slightly more expensive. And there are a lot

Reply:
Sorry, my bad. The idea was to keep backward compatibility but it's obviously not working like this. I still think it's possible to do it with a computed property, converting the camel caps with regex i.e. Will fix it and submit.